### PR TITLE
add the UniPC scheduler

### PR DESCRIPTION
**Pull request by wl-zhao**
Add a multistep UniPC. UniPC is a training-free framework designed for the fast sampling of diffusion models, which consists of a corrector (UniC) and a predictor (UniP) that share a unified analytical form and support arbitrary orders.

For more details about the method, please refer to the [[paper]](https://arxiv.org/abs/2302.04867) and the [[code]](https://github.com/wl-zhao/UniPC).

This implementation also supports passing another scheduler as the `solver_p` to UniPC, which will instead perform the sampling through `solver_p + UniC`.

https://github.com/huggingface/diffusers/issues/2311

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Very cool the scheduler works very well:

```python
#!/usr/bin/env python3
from diffusers import StableDiffusionPipeline, UniPCMultistepScheduler
import time
import torch
import sys

path = "CompVis/stable-diffusion-v1-4"

pipe = StableDiffusionPipeline.from_pretrained(path, torch_dtype=torch.float16)
pipe.scheduler = UniPCMultistepScheduler.from_config(pipe.scheduler.config)
pipe = pipe.to("cuda")

prompt = "an image of Santa Claus in light blue"
generator = torch.Generator(device="cuda").manual_seed(0)
image = pipe(prompt, generator=generator, num_inference_steps=25).images[0]
```

aa
